### PR TITLE
Add enable_zsa flag

### PR DESCRIPTION
**Pull request by ConstanceBeguier**
This PR adds enable_zsa flag into ZIP 226.

### CODE REVIEW

**Review by vivek-arte — COMMENTED** [review]
We need to add the `enable_zsa` boolean into the Transaction Structure as well.

I think the most efficient place is to reserve another bit in `flagsOrchard` (it currently only has `enableSpendsOrchard` and `enableOutputsOrchard`) for `enable_zsa`? I believe that is what has been done by @ConstanceBeguier in the [orchard PR](https://github.com/QED-it/orchard/pull/79).

The only point of contention is calling it `flagsOrchard` then, when there is a ZSA specific component inside it. But changing it to `flagsOrchardZSA` also seems like something that could be easy to cause issues...

@PaulLaux

**Comment by vivek-arte** [comment]
Note that I also have to include `enable_zsa` into the v6 tx structure [PR](https://github.com/QED-it/zips/pull/36)

File: zip-0226.rst [hunk 1]
```diff
@@ -60,6 +60,8 @@ Because in a single transaction all value commitments are balanced, there must b
 
 In order to ensure the security of the transfers, and as we will explain below, we are redefining input dummy notes [#protocol-dummynotes]_ for Custom Assets, as we need to enforce that the :math:`\mathsf{AssetBase}^{\mathsf{Orchard}}` of the output note of that Split Action is the output of a valid :math:`\mathsf{ZSAValueBase^{Orchard}}` computation defined in ZIP 227 [#zip-0227]_.
 
+In order to be able to pause the ZSA functionality, we added the :math:`\mathsf{enable\_zsa}` boolean flag. When this flag is set to false, it is not possible to perform ZSA transactions (the circuit will fail).
```
[the remark this answers is not in the document]
Comment (ConstanceBeguier) [reply zip-0226.rst]: Done

File: zip-0226.rst [hunk 2]
```diff
@@ -247,6 +249,13 @@ The following constraints must be added to ensure that the input and output note
 
 To make the evaluation of the note commitment easier, we add a boolean :math:`\mathsf{is\_native\_asset}` as an auxiliary witness. We also add some constraints to verify that this variable is activated (i.e. :math:`\mathsf{is\_native\_asset} = 1`) if the Asset Base is equal to :math:`\mathcal{V}^\mathsf{Orchard}` and this variable is not activated (i.e. :math:`\mathsf{is\_native\_asset} = 0`) if the Asset Base is not equal to :math:`\mathcal{V}^\mathsf{Orchard}`.
 
+enable\_zsa Flag
+````````````````
+
+The following constraints must be added to disable ZSA transactions when the :math:`\mathsf{enable\_zsa}` flag is set to false:
+
+- if :math:`\mathsf{enable\_zsa}` is not activated (i.e. :math:`\mathsf{enable\_zsa} = 0`), then the :math:`\mathsf{AsssetBase}` must be equal to the native asset (i.e. :math:`\mathsf{is\_native\_asset} = 1`).
```
[the remark this answers is not in the document]
Comment (ConstanceBeguier) [reply zip-0226.rst]: Done

File: zip-0226.rst [hunk 3]
```diff
@@ -247,6 +249,13 @@ The following constraints must be added to ensure that the input and output note
 
 To make the evaluation of the note commitment easier, we add a boolean :math:`\mathsf{is\_native\_asset}` as an auxiliary witness. We also add some constraints to verify that this variable is activated (i.e. :math:`\mathsf{is\_native\_asset} = 1`) if the Asset Base is equal to :math:`\mathcal{V}^\mathsf{Orchard}` and this variable is not activated (i.e. :math:`\mathsf{is\_native\_asset} = 0`) if the Asset Base is not equal to :math:`\mathcal{V}^\mathsf{Orchard}`.
 
+enable\_zsa Flag
+````````````````
+
```
[the remark this answers is not in the document]
Comment (ConstanceBeguier) [reply zip-0226.rst]: Done

File: zip-0226.rst [hunk 4]
```diff
@@ -247,6 +249,13 @@ The following constraints must be added to ensure that the input and output note
...
+The following constraints must be added to disable ZSA transactions when the :math:`\mathsf{enable\_zsa}` flag is set to false:
```
[the remark this answers is not in the document]
Comment (ConstanceBeguier) [reply zip-0226.rst]: Done

**Comment by ConstanceBeguier** [comment]
> We need to add the `enable_zsa` boolean into the Transaction Structure as well.
> 
> I think the most efficient place is to reserve another bit in `flagsOrchard` (it currently only has `enableSpendsOrchard` and `enableOutputsOrchard`) for `enable_zsa`? I believe that is what has been done by @ConstanceBeguier in the [orchard PR](https://github.com/QED-it/orchard/pull/79).
> 
> The only point of contention is calling it `flagsOrchard` then, when there is a ZSA specific component inside it. But changing it to `flagsOrchardZSA` also seems like something that could be easy to cause issues...
> 
> @PaulLaux

Yes, I added the enable ZSA flag in flagsOrchard.

**Comment by vivek-arte** [comment]
image

In the ZSA Transaction Structure section of ZIP 226, we could just add the details of enable_zsa flag in the same manner as currently done for enableSpendsOrchard and enableOutputsOrchard.

There is a column for descriptions that is currently blank for flagsOrchard, since there weren't changes earlier. We should add the details into that cell now, same as in the spec, with an additional line for enable_zsa

**Comment by vivek-arte** [comment]
Also, for consistency with enableSpendsOrchard and enableOutputsOrchard, in the spec we should use enableZSA instead of enable_zsa

**Comment by ConstanceBeguier** [comment]
> Also, for consistency with enableSpendsOrchard and enableOutputsOrchard, in the spec we should use enableZSA instead of enable_zsa

Done

**Review by PaulLaux — APPROVED** [review]

**Review by PaulLaux — APPROVED** [review]
Please revert changes to zipp 225 (not in the scope of the zsa project) and merge

**Pull request merged**